zebra-rpc/src/methods/get_block_template_rpcs.rs

```diff
@@ -346,9 +379,18 @@ where
 
                 coinbase_txn: TransactionTemplate::from_coinbase(&coinbase_tx, miner_fee),
 
-                target: empty_string.clone(),
+                target: format!(
+                    "{}",
+                    expected_difficulty
+                        .to_expanded()
+                        .ok_or(Error {
+                            code: ErrorCode::ServerError(0),
+                            message: "Not enough blocks in the chain".to_string(),
+                            data: None,
+                        })?
```

Comment:
The state should never return an invalid difficulty, because it validates blocks before committing them:
```suggestion
                        .expect("state always returns a valid difficulty value")
```

Reply:
What is the worst it can happen if we just leave this error as is ?

Reply:
This error should never happen, so it is ok to merge it.